src/main/scala/mesosphere/mesos/Availability.scala

```diff
@@ -0,0 +1,23 @@
+package mesosphere.mesos
+
+import org.apache.mesos.Protos.Offer
+
+object Availability {
+
+  def offerAvailable(offer: Offer): Boolean = {
+    var unavailability = offer.hasUnavailability
+    var now = System.nanoTime()
+    if (offer.hasUnavailability && offer.getUnavailability.hasStart) {
+      val start = offer.getUnavailability.getStart.getNanoseconds
+      if (now.>=(start)) {
+        if (offer.getUnavailability.hasDuration) {
+          return start.+(offer.getUnavailability.getDuration.getNanoseconds).<(now)
```

Comment:
Hmm...as far as I understand maintenance mode, the agent is supposed to be offline during the scheduled maintenance frame (time between `start` and `start + duration`). If the agent is stopped to do this maintenance, the agent will not send offers.
To have a smooth behavior, we should implement this feature in multiple steps:
1) Ignore offers upfront (for a configurable amount of time) to the scheduled downtime (and during as well, if the agent is not down as planned), as @janisz suggested. I think this step could be easily added to this pr.
2) Implement a shift, where marathon moves all tasks from the to-be-maintained agent away to other agents. In the best case the start of this shift also should be configurable.